units/provisioning_create_host.go

```diff
@@ -47,20 +50,34 @@ func makeCreateHostJob() *createHostJob {
 	return j
 }
 
-func NewHostCreateJob(env evergreen.Environment, h host.Host, id string) amboy.Job {
+func NewHostCreateJob(env evergreen.Environment, h host.Host, id string, CurrentAttempt int, MaxAttempts int) amboy.Job {
```

Comment:
I think it's really easy (and not clear from the call site) to flip successive integer elements. Does it make sense to define a (util) "RetryCounter" type.